prov/verbs/src/verbs_info.c

```diff
@@ -361,7 +361,8 @@ int fi_ibv_check_rx_attr(const struct fi_rx_attr *attr,
 		return -FI_ENODATA;
 	}
 
-	if (attr->total_buffered_recv > info->rx_attr->total_buffered_recv) {
+	if (!attr->total_buffered_recv &&
+		(attr->total_buffered_recv > info->rx_attr->total_buffered_recv)) {
```

Comment:
I'm not sure if I understand your comment. If a provider is able to receive unexpected messages without losing them, then that means the provider supports FI_RM_ENABLED. The app can choose to disable RM if it wants.

If the app requests a total_buffered_recv > available buffering, we have to return an error. The change in the PR allows this check only when total_buffered_recv is 0.